### PR TITLE
Improve system setup 

### PR DESCRIPTION
**Pull request by Stadicus**
This pull requests aims at improving the system build and initialization process:
* use dedicated scripts to start main applications from systemd
* improve permission handling in `.bitcoin` directory
* add additional metrics like hostname or build information to Prometheus
* extend Grafana dashboard with these metrics
* general code cleanup and additional components

### CODE REVIEW

File: armbian/base/scripts/systemd-startup-checks.sh [hunk 1]
```diff
@@ -56,20 +56,22 @@ fi
...
-chmod -R 750 /mnt/ssd
 
-# We set rpccookiefile=/mnt/ssd/bitcoin/.bitcoin/.cookie, but there seems to be
-# no way to specify where to expect the bitcoin cookie for c-lightning, so let's
-# create a symlink at the expected testnet location.
-mkdir -p /mnt/ssd/bitcoin/.bitcoin/testnet3/
-ln -fs /mnt/ssd/bitcoin/.bitcoin/.cookie /mnt/ssd/bitcoin/.bitcoin/testnet3/.cookie
+# set permissions for whole ssd 
+# (user:rwx group:r-x other:---)
+chmod -R 750 /mnt/ssd
```
Comment (hkjn) [inline armbian/base/scripts/systemd-startup-checks.sh]: Nit: Missing newline at end of file.

File: armbian/base/scripts/prometheus-base.py [hunk 2]
```diff
@@ -5,13 +5,19 @@
 import time
 import subprocess
 import sys
-from prometheus_client import start_http_server, Gauge, Counter
+from prometheus_client import start_http_server, Gauge, Counter, Info
+
+SYSCONFIG_PATH="/opt/shift/sysconfig/"
 
 # Create Prometheus metrics to track bitcoind stats.
```
Comment (hkjn) [inline armbian/base/scripts/prometheus-base.py]: (Optional) Not new in this PR, but perhaps this comment would be more accurate if written as `Create Prometheus metrics to track Base stats.`?

File: armbian/base/scripts/prometheus-base.py [hunk 3]
```diff
@@ -53,4 +68,4 @@ def main():
 
 
 if __name__ == "__main__":
-    main()
+    main()
```
Comment (hkjn) [inline armbian/base/scripts/prometheus-base.py]: Nit: Missing newline at end of file.

File: armbian/base/scripts/systemd-start-bitcoind.sh [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+/usr/bin/bitcoind -daemon -conf=/etc/bitcoin/bitcoin.conf
+
+# wait a few seconds before providing cookie authentication 
+# as .env file for electrs and base-middleware 
+sleep 10
+echo -n 'RPCPASSWORD=' > /mnt/ssd/bitcoin/.bitcoin/.cookie.env
+tail -c +12 /mnt/ssd/bitcoin/.bitcoin/.cookie >> /mnt/ssd/bitcoin/.bitcoin/.cookie.env
+
+# hold off next services for a bit
+sleep 10
```
Comment (hkjn) [inline armbian/base/scripts/systemd-start-bitcoind.sh]: Nit: Missing newline at end of file.

File: armbian/base/scripts/systemd-start-electrs.sh [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+source /mnt/ssd/bitcoin/.bitcoin/.cookie.env
+
+# start main application
+/usr/bin/electrs \
+    --network ${NETWORK} \
+    --db-dir ${DB_DIR} \
+    --daemon-dir ${DAEMON_DIR} \
+    --cookie "__cookie__:${RPCPASSWORD}" \
+    --monitoring-addr ${MONITORING_ADDR} \
+    -${VERBOSITY}
```
Comment (hkjn) [inline armbian/base/scripts/systemd-start-electrs.sh]: Nit: Missing newline at end of file.

File: armbian/base/scripts/systemd-start-bitcoind.sh [hunk 6]
```diff
@@ -0,0 +1,22 @@
+#!/bin/bash
```
Comment (hkjn) [inline armbian/base/scripts/systemd-start-bitcoind.sh]: (Optional) Should we consider adding some options to make these scripts more robust? I.e `set -euo pipefail`? (And elsewhere.)

File: armbian/base/scripts/systemd-start-bitcoind.sh [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+
```
Comment (hkjn) [inline armbian/base/scripts/systemd-start-bitcoind.sh]: (Optional) Could we add a brief comment explaining what these scripts do? Maybe something like:

```
#!/bin/bash
#
# This script is called by the bitcoind.service to start bitcoind.
#
```

(And elsewhere.)

File: armbian/base/scripts/systemd-start-bitcoind.sh (hunk 6 above)
Comment (Stadicus) [reply armbian/base/scripts/systemd-start-bitcoind.sh]: Good point, I'll do that in the next iteration.

**Comment by Stadicus** [comment]
Thanks for the review, all points have been addressed.

**Pull request merged**